src/families/cosmos/libcore-buildTransaction.js

```diff
@@ -65,28 +65,32 @@ export async function cosmosBuildTransaction({
   await transactionBuilder.setMemo(memoTransaction);
 
   // Gas
-  let gas: BigNumber;
+  let estimatedGas: BigNumber;
 
-  if (gasLimit && gasLimit !== "0") {
-    gas = BigNumber(gasLimit);
+  if (gas && gas !== "0") {
+    estimatedGas = BigNumber(gas);
   } else {
     const gasRequest = await core.CosmosGasLimitRequest.init(
       memoTransaction,
       messages,
       getEnv("COSMOS_GAS_AMPLIFIER")
     );
-    gas = await libcoreBigIntToBigNumber(
+    estimatedGas = await libcoreBigIntToBigNumber(
       await cosmosLikeAccount.estimateGas(gasRequest)
```

Comment:
why are we estimating again the gas? i think it can still mismatch if we do this?

Reply:
Well we estimated here on the first time we call the buildTransaction, then on the second time the gas should be set in the Transaction object if it's go by prepareTransaction.

Reply:
but then, wouldn't this produce the actual bug we are currently having where the device does not match the Live display?

Reply:
OH i see it now!